hass_nabucasa/__init__.py

```diff
@@ -136,7 +136,8 @@ def subscription_expired(self) -> bool:
     def expiration_date(self) -> datetime:
         """Return the subscription expiration as a UTC datetime object."""
         return datetime.combine(
-            parse_date(self.claims["custom:sub-exp"]), datetime.min.time()
+            parse_date(self.claims["custom:sub-exp"]) or datetime.now().date(),
```

Comment:
If I recall correctly, this is mainly used to display the data in the UI. Alternatively, this method could be adjusted from `datetime` to `datetime | None` as well; the UI could show "Unknown" in that case (when `None`).

Reply:
Did that 👍 

Currently it would raise `TypeError` if `parse_date(self.claims["custom:sub-exp"])` is None, and that never happens, so this change is not really changing anything as we never get here, but this makes more logical sense and it does not bite us later if something changes